mne/bem.py

```diff
@@ -0,0 +1,321 @@
+    u['w'][-1] = 0
+
+    # Do the nonlinear minimization, constraining mu to the interval [-1, +1]
+    mu_0 = np.random.RandomState(0).rand(nfit) * f
+    fun = partial(_one_step, u=u)
+    cons = []
+    for ii in range(nfit):
+        for val in [1., -1.]:
+            cons.append({'type': 'ineq',
+                         'fun': lambda x: np.array([val * x[ii] + 1.]),
+                         'jac': lambda x: np.array([0.] * ii + [val] +
+                                                   [0.] * (nfit - ii - 1))})
+    mu = optimize.minimize(fun, mu_0, constraints=cons, method='COBYLA',
+                           tol=1e-2).x
+
+    # (6) Do the final step: calculation of the linear parameters
+    rv, lambda_ = _compute_linear_parameters(mu, u)
+    order = np.argsort(mu)[::-1]
+    mu, lambda_ = mu[order], lambda_[order]  # sort: largest mu first
+
+    m['mu'] = mu
+    # This division takes into account the actual conductivities
+    m['lambda'] = lambda_ / m['layers'][-1]['sigma']
+    m['nfit'] = nfit
+    return rv
+
+
+@verbose
+def make_sphere_model(r0=(0., 0., 0.04), head_radius=0.09, info=None,
+                      verbose=None):
+    """Create a spherical model for forward solution calculation
+
+    Parameters
+    ----------
+    r0 : array-like | str
+        Head center to use (in head coordinates). If 'auto', the head
+        center will be calculated from the digitization points in info.
+    head_radius : float | str | None
+        If float, compute spherical shells for EEG using the given radius.
+        If 'auto', estimate an approriate radius from the dig points in Info,
+        If None, exclude shells.
+    info : instance of mne.io.meas_info.Info | None
+        Measurement info. Only needed if ``r0`` or ``head_radius`` are
+        ``'auto'``.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    bem : dict
+        A spherical BEM.
+    """
+    for name in ('r0', 'head_radius'):
+        param = locals()[name]
+        if isinstance(param, string_types):
+            if param != 'auto':
+                raise ValueError('%s, if str, must be "auto" not "%s"'
+                                 % (name, param))
+
+    if (isinstance(r0, string_types) and r0 == 'auto') or \
+       (isinstance(head_radius, string_types) and head_radius == 'auto'):
+        if info is None:
+            raise ValueError('Info must not be None for auto mode')
+        head_radius_fit, r0_fit = fit_sphere_to_headshape(info)[:2]
+        if isinstance(r0, string_types):
+            r0 = r0_fit
+        if isinstance(head_radius, string_types):
+            head_radius = head_radius_fit
+    sphere = dict(r0=np.array(r0), is_sphere=True,
+                  coord_frame=FIFF.FIFFV_COORD_HEAD)
+    sphere['layers'] = []
+    if head_radius is not None:
+        # Eventually these could be configurable...
```

Comment:
The radii and sigmas? Are these often changed by people? Note that the radii here are relative to the actual radius, which is already a parameter the user can change.